lib/bpf/bpf.go

```diff
@@ -205,13 +205,23 @@ func (s *Service) OpenSession(ctx *SessionContext) (uint64, error) {
 
 	// initializedModClosures holds all already opened modules closures.
 	initializedModClosures := make([]sessionEnder, 0)
-	for _, module := range []cgroupRegister{
-		s.open,
-		s.exec,
-		s.conn,
+	for _, m := range []struct {
+		eventName string
+		module    cgroupRegister
+	}{
+		{constants.EnhancedRecordingCommand, s.exec},
+		{constants.EnhancedRecordingDisk, s.open},
+		{constants.EnhancedRecordingNetwork, s.conn},
 	} {
+		// If the event is not being monitored in this session we
+		// shouldn't start monitoring it.
+		_, ok := ctx.Events[m.eventName]
+		if !ok {
```

Comment:
Suggestion: reduce the scope of ok.
```suggestion
		if _, ok := ctx.Events[m.eventName]; !ok {
```